### PR TITLE
Hw 03

### PR DESCRIPTION
**Pull request by andreikasmykou**

### CODE REVIEW

File: src/utils/streams.js [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+const customHelp = () => {
+  console.log('---Read help documentation and try again please---'); // eslint-disable-line no-console
+};
+
+//Main functions
+/**
+ * @description Reverse string data from process.stdin to process.stdout
+ * @param {string} str (first data for reversing)
+ * @return void
+ */
```
Comment (slowpokex) [inline src/utils/streams.js:40]: Docs isn't nessessary, but good practice

File: src/utils/streams.js [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+/**
+ * @description Create Transform Stream for reading given URL, collecting all data and joining it
+ * to the data from the stream which invoke this transformer
+ * @param {string} URL  (resource from which need add data)
+ * @return {Stream}
+ */
+const addDataFromUrl = (URL) => {
+  return through(
+    (data, enc, cb) => {
+      request(URL, (error, response, body) => {
```
Comment (slowpokex) [inline src/utils/streams.js]: What about error handling?

Comment (slowpokex) [reply src/utils/streams.js]: And you can make it with pipe
```
request('http://google.com/doodle.png').pipe(fs.createWriteStream('doodle.png'))
```

File: src/utils/streams.js [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+//Main functions
+/**
+ * @description Reverse string data from process.stdin to process.stdout
+ * @param {string} str (first data for reversing)
+ * @return void
+ */
+export const reverse = (str = '') => {
+  process.stdin.on('readable', () => {
+    const chunk = process.stdin.read();
+    if (chunk !== null) {
```
Comment (slowpokex) [inline src/utils/streams.js]: If we have undefined case, what should I see?

File: src/utils/streams.js [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+/**
+ * @description Read given file provided by --file option and pipe it to process.stdout.
+ * @param {string} filePath
+ * @return void
+ */
+export const outputFile = (filePath) => {
+  if (isFile(filePath)) {
+    fs.createReadStream(filePath)
+      .pipe(process.stdout);
+  }
```
Comment (slowpokex) [inline src/utils/streams.js]: If we have directory or wrong file?

File: src/utils/streams.js [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+
+/**
+ * @description Convert file provided by --file option from csv to json,
+ * using the through2 module, and output data to process.stdout.
+ * Function check that the passed file name is valid.
+ * @param {string} filePath
+ * @return void
+ */
+export const convertFromFile = (filePath) => {
+  if (isFile(filePath)) {
```
Comment (slowpokex) [inline src/utils/streams.js:95]: Here too

File: src/utils/streams.js [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+ *  3. Add contents of
+ *  https://drive.google.com/uc?export=download&id=1tCm9Xb4mok4Egy2WjGqdYYkrGia0eh7X
+ *  at the bottom of this big css
+ *  4. Output will be saved in the same path and called bundle.css
+ * @param {string} directoryPath (extra parameter --path in CL)
+ * @throws {WrongPathError}
+ * @return void
+ */
+export const makeCssBundle = (directoryPath) => {
+  if (isDirectory(directoryPath)) {
```
Comment (slowpokex) [inline src/utils/streams.js:141]: Please make another scenario, like if we have a file instead of directory

File: src/utils/streams.js [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+ * @description Return true if given filePath is File
+ * @param {string} filePath
+ * @return {boolean}
+ */
+const isFile = (filePath) => {
+  if (fs.statSync(filePath).isFile()) {
+    return true;
+  }
+  console.log(`Wrong filePath: ${filePath}`); // eslint-disable-line no-console
+  program.help();
```
Comment (slowpokex) [inline src/utils/streams.js]: please return false if it not file

File: src/utils/streams.js [hunk 8]
```diff
@@ -0,0 +1,256 @@
...
+ * @description Return true if given directoryPath is Directory
+ * @param {string} directoryPath
+ * @return {boolean}
+ */
+const isDirectory = (directoryPath) => {
+  if (fs.statSync(directoryPath).isDirectory()) {
+    return true;
+  }
+  console.log(`Wrong directoryPath: ${directoryPath}`); // eslint-disable-line no-console
+  program.help();
```
Comment (slowpokex) [inline src/utils/streams.js]: Here too

**Review by slowpokex — CHANGES_REQUESTED** [review]

File: src/utils/streams.js (hunk 1 above)
Comment (andreikasmykou) [reply src/utils/streams.js:40]: Ok)

File: src/utils/streams.js (hunk 2 above)
Comment (andreikasmykou) [reply src/utils/streams.js]: Forgot about error. My fault(
About piping, we need to add data from request to the end file, I tried, but this approach replace all data.

File: src/utils/streams.js (hunk 4 above)
Comment (andreikasmykou) [reply src/utils/streams.js]: Refactor a little, now user will see message and help info

File: src/utils/streams.js (hunk 5 above)
Comment (andreikasmykou) [reply src/utils/streams.js:95]: Same

File: src/utils/streams.js (hunk 6 above)
Comment (andreikasmykou) [reply src/utils/streams.js:141]: If it is not a directory user will get notification that directoryPath is wrong and help info

File: src/utils/streams.js (hunk 7 above)
Comment (andreikasmykou) [reply src/utils/streams.js]: Done

File: src/utils/streams.js (hunk 8 above)
Comment (andreikasmykou) [reply src/utils/streams.js]: Done

File: src/utils/streams.js (hunk 3 above)
Comment (andreikasmykou) [reply src/utils/streams.js]: Add the same logic as for null

**Review by slowpokex — APPROVED** [review]
Looks good

**Pull request merged**